File: src/main/scala/almoneya/BankAccountTransaction.scala
package almoneya

import org.joda.time.LocalDate

case class BankAccountTransaction(id: Option[BankAccountTransactionId] = None,
                                  bankAccount: BankAccount,
                                  checkNum: Option[CheckNum] = None,
                                  postedOn: LocalDate,
                                  description1: Option[Description] = None,
                                  description2: Option[Description] = None,
                                  amount: Amount)

File: lib/repositories/bank_account_transaction.rb
module Repositories
  BankAccountTransaction = Struct.new(:bank_account_transaction_id, :account_num, :check_num, :posted_on, :description1, :description2, :amount, :created_at, :updated_at) do
    alias_method :id, :bank_account_transaction_id
  end
end
